### DISCUSSION TITLE
WebAssembly Support

### DISCUSSION THREAD

**Discussion by sedbytes** [explosion/spaCy#12562] [New Features & Project Ideas]
I wonder if its possible to compile spaCy into a browser-accessible wasm interface. This would allow all the computations to occur client-side and eliminate the need for a backend server (like in spacy.js).

If it is possible, did someone ever get it to work?

I looked for this everywhere, can't find anything appropriate.

**Comment by sedbytes** [comment]
Update: I got it to work using [pyodide](https://github.com/pyodide/pyodide). Here's an example: https://github.com/SyedAhkam/spacy-wasm

Although It should be noted that I had to build spacy for wasm using [exodide](https://github.com/ymd-h/exodide). I made the sample wheels available in the `packages` directory.

**Comment by sedbytes** [comment]
I'll keep this open since pyodide isn't the officially supported way.

**Comment by rmitsch** [comment]
Hi @SyedAhkam, that's a really cool project, thanks for sharing! We haven't looked into that yet AFAIK, let me get back to you in a bit.

**Reply by rmitsch** [reply]
We'd like to feature `spacy-wasm` in the [spaCy universe](https://spacy.io/universe) and tweet about it as well, because we dig it a lot :) Would that be ok with you?

**Reply by sedbytes** [reply]
Woah. I certainly didn't expect such traction. I was mostly just trying to satisfy my curiosity haha.

However, One thing I'd like to point: The exodide repository deserves an equivalent if not more in terms of credit. Since if you were to look [here](https://github.com/ymd-h/exodide/tree/master/example-spaCy), they did have something similar going on already. Then I came around and rewrote the code to simplify and make things more intuitive.

Now taking that outta the way, I would be more than happy if you wish to feature / tweet about it. You just made my day :)

PS: My twitter is `SyedAhkam1`

**Reply by rmitsch** [reply]
Thanks for letting us know about that! We'll make sure to acknowledge the `exodide` team as well :-) I'll update here once the tweet is out and/or the spaCy universe entry has been added (the latter you can also do yourself, if you want to - see [here](https://github.com/explosion/spaCy/pull/12568) for a recent PR example on this).

**Reply by rmitsch** [reply]
[Twitter](https://twitter.com/spacy_io/status/1651545295879122944?s=20) - [Mastodon](https://sigmoid.social/@explosion/110270482736091657) - [LinkedIn](https://www.linkedin.com/feed/update/urn:li:activity:7057311217702334464) :slightly_smiling_face: 
The spaCy universe entry has been merged.

**Reply by sedbytes** [reply]
Tysm!! I'm overjoyed by the community response that I'm receiving.

Now looking forward, It'd be great if you guys actually look into optimisation of my approach and build an official solution which would provide a completely client-side interface to spaCy. Perhaps spacy.js reborn or spacy.wasm? There's so many possibilities.

Hoping for a bright future at spaCy and explosion 💥

**Reply by Blbns2042** [reply]
Awesome. is there a spacy.wasm/spacy.js ready for use?

**Comment by MRYingLEE** [comment]
FYI. I just found newly published wheels for Spacy at https://github.com/liu-nlp/spacy-pyodide. Although some issue blocked me to install, maybe the author will clarify soon.
